### PR TITLE
Initial main

### PR DESCRIPTION
**Pull request by nlake44**

### CODE REVIEW

File: main/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Uage:
```
Comment (jovanchohan) [inline main/Makefile]: Usage*

File: main/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# make	 	# compile all
```
Comment (jovanchohan) [inline main/Makefile]: Spacing of comments could be lined up

File: main/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# make clean	# remote ALL binaries and objects
+
+# This is an example comment.
+
+# Defind compile to be used.
```
Comment (jovanchohan) [inline main/Makefile]: What is 'Defind'?

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
 .DS_Store
+*.o
```
Comment (brandonin) [inline .gitignore:6]: I would also include just a plain boilerplate .gitignore. See https://github.com/github/gitignore, or more specifically: https://github.com/github/gitignore/blob/master/C.gitignore

**Comment by brandonin** [comment]
See https://github.com/supragma/supragma/pull/12 for review of the duplicated portions.

File: doc/README.md [hunk 5]
```diff
@@ -1,7 +1,29 @@
 # Installation for OSX Development
+TODO(raj): Write a script to do all this
```
Comment (brandonin) [inline doc/README.md]: We wouldn't necessarily need to do it for a specific development environment if it is dockerized.

File: main/Makefile [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+CC=gcc
+
+# Name of final binary.
+TARGET=supragma
+ 
+all:    main.o
+	$(CC) main.c -o $(TARGET)
+ 
+clean:
+	rm *.o $(TARGET)
```
Comment (brandonin) [inline main/Makefile]: Maybe use $(RM) instead of rm

Comment (brandonin) [reply main/Makefile]: https://www.cs.swarthmore.edu/~newhall/unixhelp/howto_makefiles.html

Comment (brandonin) [reply main/Makefile]: Follows the same paradigm as the all command.

File: main/Makefile [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# Usage:
+# make          # compile all
+# make clean    # remote ALL binaries and objects
+
+# This is an example comment.
```
Comment (brandonin) [inline main/Makefile]: I think this can be derived from all of the other comments.

**Review by jovanchohan — APPROVED** [review]

File: .gitignore (hunk 4 above)
Comment (brandonin) [reply .gitignore:6]: resolved.

File: doc/README.md (hunk 5 above)
Comment (nlake44) [reply doc/README.md]: resolved to containerization TODO

File: main/Makefile (hunk 6 above)
Comment (nlake44) [reply main/Makefile]: Resolved

File: main/Makefile (hunk 7 above)
Comment (nlake44) [reply main/Makefile]: Resolved

**Pull request merged**